src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -205,6 +213,27 @@ async Task UpdateUnderLockAsync(UnconfiguredProject project, Action<ProjectRootE
             }
         }
 
+        public virtual Task<IEnumerable<KeyValuePair<string, IEnumerable<string>>>> GetProjectConfigurationDimensionsAsync(Project project)
+        {
+            Requires.NotNull(project, nameof(project));
+
+            ImmutableArray<string> values = GetOrderedPropertyValues(project);
+            if (values.IsEmpty)
+            {
+                return Task.FromResult(Enumerable.Empty<KeyValuePair<string, IEnumerable<string>>>());
+            }
+            else
+            {
+                KeyValuePair<string, IEnumerable<string>>[]? dimensionValues = new[] { new KeyValuePair<string, IEnumerable<string>>(DimensionName, values) };
+                return Task.FromResult(dimensionValues.AsEnumerable());
```

Comment:
Similar to the above, this can be done with compile-time hinting rather than run-time work.

```suggestion
                IEnumerable<KeyValuePair<string, IEnumerable<string>>> dimensionValues = new[] { new KeyValuePair<string, IEnumerable<string>>(DimensionName, values) };
                return Task.FromResult(dimensionValues);
```

or

```suggestion
                KeyValuePair<string, IEnumerable<string>>[] dimensionValues = new[] { new KeyValuePair<string, IEnumerable<string>>(DimensionName, values) };
                return Task.FromResult<IEnumerable<KeyValuePair<string, IEnumerable<string>>>>(dimensionValues);
```

Reply:
Done